### PR TITLE
Make request timeout customizable in `HttpArtifactRepository`

### LINKED ISSUE #6202: Timeout value too small: when downloading large model files, timeout is reached
https://github.com/mlflow/mlflow/blob/d40780be361f4bd2741c2e8fcbd428c1d693edcf/mlflow/store/artifact/http_artifact_repo.py#L63

### LINKED ISSUE #6205: [BUG] Http artifact repo time out for large artifact downloads
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.27.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: docker
- **Python version**: 3.7
- **yarn version, if running the dev UI**:

### Describe the problem

The default timeout for downloading artifacts via http artifact store is 10seconds (https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/http_artifact_repo.py#L63-L72).

This causes timeout errors if the server takes longer than 10 seconds to serve the artifact (may happen for large file downloads)

Suggest to set this as a keyword argument so that we can set a timeout for mlflow.pyfunc.load_model for large file downloads that will may take more than 10s for a response.

### Tracking information

_No response_

### Code to reproduce issue

```python
import mlflow

mlflow.pyfunc.load_model("runs:/xxx/xxx")
```

If the model .pkl file is very large, it will cause urllib to throw timeout errors with every retry due to the hard-coded 10s timeout.

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

Fix #6205
Fix #6202 

## What changes are proposed in this pull request?

Make request timeout customizable in `HttpArtifactRepository`.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/http_artifact_repo.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def log_artifact(self, local_file, artifact_path=None):
         paths = (artifact_path, file_name) if artifact_path else (file_name,)
         endpoint = posixpath.join("/", *paths)
         with open(local_file, "rb") as f:
-            resp = http_request(self._host_creds, endpoint, "PUT", data=f, timeout=600)
+            resp = http_request(self._host_creds, endpoint, "PUT", data=f)
```
Comment (harupy) [inline mlflow/store/artifact/http_artifact_repo.py:25]: Maybe we should raise a warning message that explains how to customize the timeout length (e.g. `API request timed out. Set environment variable MLFLOW_HTTP_REQUEST_TIMEOUT (default: 120) to increase timeout.`) in `http_request` if it fails due to timeout.

Comment (dbczumar) [reply mlflow/store/artifact/http_artifact_repo.py:25]: Sounds like a good idea to me :)

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: mlflow/store/artifact/http_artifact_repo.py (hunk 1 above)
Comment (WeichenXu123) [reply mlflow/store/artifact/http_artifact_repo.py:25]: Thought: 
Does a global default `MLFLOW_HTTP_REQUEST_TIMEOUT` satisfy all cases ?
e.g.  downloading model we should set timeout a bit higher. But others we don't need so long timeout.

Comment (harupy) [reply mlflow/store/artifact/http_artifact_repo.py:25]: It's probably not ideal to use a single timeout value for all the operations, but I think it's ok.

Comment (harupy) [reply mlflow/store/artifact/http_artifact_repo.py:25]: @dbczumar @WeichenXu123 Filed a follow-up PR: https://github.com/mlflow/mlflow/pull/6208

**Pull request merged**